app/helpers/application_helper.rb

```diff
@@ -343,12 +343,27 @@ def name_to_id(name)
     name.to_s.gsub(/\]\[|[^-a-zA-Z0-9:.]/, "_").sub(/_$/, "")
   end
 
-  def field_id(form, attribute)
-    name_to_id(field_name(form, attribute))
+  def field_id(form_or_object_name, attribute, index: nil, **_kwargs)
+    name_to_id(field_name(form_or_object_name, attribute, index: index))
   end
 
-  def field_name(form, attribute)
-    "#{form.object_name}[#{field_attribute(attribute)}]"
+  # This is a partial re-implementation of ActionView::Helpers::FormTagHelper#field_name.
+  # The method contract changed in Rails 7.0, but we can't use the default because it sometimes
+  # includes other information that -- at a minimum -- wrecks havoc on the Cucumber feature tests.
+  # It is used in when constructing forms, like in app/views/tags/new.html.erb.
```

Comment:
I think I'm missing something... where in that file is it used?

Reply:
It's not very direct, but calls like https://github.com/otwcode/otwarchive/blob/68dff8e5164b3b14b3b4b574a63b379d44ab2f57/app/views/tags/new.html.erb#L17 eventually make their way to this method